Source/WTF/wtf/text/StringCommon.h

```diff
@@ -70,72 +71,107 @@ bool equalLettersIgnoringASCIICase(const char*, ASCIILiteral);
+        if (unalignedLoad<uint64_t>(aLChar) != unalignedLoad<uint64_t>(bLChar))
             return false;
-
-        a += sizeof(uint16_t);
-        b += sizeof(uint16_t);
+        for (unsigned i = length % 8; i < length; i += 8) {
+            if (unalignedLoad<uint64_t>(aLChar + i) != unalignedLoad<uint64_t>(bLChar + i))
+                return false;
+        }
+        return true;
+#endif
     }
-
-    if (length & 1 && (*reinterpret_cast<const LChar*>(a) != *reinterpret_cast<const LChar*>(b)))
-        return false;
-
-    return true;
 }
 
 ALWAYS_INLINE bool equal(const UChar* aUChar, const UChar* bUChar, unsigned length)
 {
-    unsigned dwordLength = length >> 2;
-
-    const char* a = reinterpret_cast<const char*>(aUChar);
-    const char* b = reinterpret_cast<const char*>(bUChar);
+    if (!length)
+        return true;
+    if (length == 1)
+        return *aUChar == *bUChar;
 
-    if (dwordLength) {
-        for (unsigned i = 0; i != dwordLength; ++i) {
-            if (unalignedLoad<uint64_t>(a) != unalignedLoad<uint64_t>(b))
+#if COMPILER(GCC_COMPATIBLE)
+    switch (sizeof(unsigned) * CHAR_BIT - clz(length - 1)) { // Works as really fast log2, since length != 0.
+#else
+    switch (fastLog2(length - 1)) {
+#endif
+    case 0:
+        RELEASE_ASSERT_NOT_REACHED();
+    case 1: // Length is 2 (4 bytes).
+        return unalignedLoad<uint32_t>(aUChar) == unalignedLoad<uint32_t>(bUChar);
+    case 2: // Length is 3 or 4 (6-8 bytes).
+        return unalignedLoad<uint32_t>(aUChar) == unalignedLoad<uint32_t>(bUChar)
+            && unalignedLoad<uint32_t>(aUChar + length - 2) == unalignedLoad<uint32_t>(bUChar + length - 2);
+    case 3: // Length is between 5 and 8 inclusive (10-16 bytes).
+        return unalignedLoad<uint64_t>(aUChar) == unalignedLoad<uint64_t>(bUChar)
+            && unalignedLoad<uint64_t>(aUChar + length - 4) == unalignedLoad<uint64_t>(bUChar + length - 4);
+#if CPU(ARM_NEON)
+    case 4: // Length is between 9 and 16 inclusive (18-32 bytes).
+        return vminvq_u16(vandq_u16(
+            vceqq_u16(unalignedLoad<uint16x8_t>(aUChar), unalignedLoad<uint16x8_t>(bUChar)),
+            vceqq_u16(unalignedLoad<uint16x8_t>(aUChar + length - 8), unalignedLoad<uint16x8_t>(bUChar + length - 8))
+        ));
+    default: // Length is longer than 16 (32 bytes).
+        if (!vminvq_u16(vceqq_u16(unalignedLoad<uint16x8_t>(aUChar), unalignedLoad<uint16x8_t>(bUChar))))
+            return false;
+        for (unsigned i = length % 8; i < length; i += 8) {
+            if (!vminvq_u16(vceqq_u16(unalignedLoad<uint16x8_t>(aUChar + i), unalignedLoad<uint16x8_t>(bUChar + i))))
                 return false;
-
-            a += sizeof(uint64_t);
-            b += sizeof(uint64_t);
         }
-    }
-
-    if (length & 2) {
-        if (unalignedLoad<uint32_t>(a) != unalignedLoad<uint32_t>(b))
+        return true;
+#else
+    default: // Length is longer than 8 (16 bytes).
+        static_assert(false);
```

Comment:
We are hitting this in GTK ports etc.

Reply:
Oh whoops, that absolutely should not still be in, it was just to confirm that `CPU(ARM_NEON)` was defined. 🤦

Reply:
Removed in latest push.